### PR TITLE
htmldoc: 1.8.29 -> 1.9.11

### PR DESCRIPTION
**Pull request by LeSuisse**
###### Motivation for this change

Fixes CVE-2021-20308.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>htmldoc</li>
  </ul>
</details>

### CODE REVIEW

File: pkgs/tools/typesetting/htmldoc/default.nix [hunk 1]
```diff
@@ -1,18 +1,16 @@
-{ lib, stdenv, fetchurl
-
-, SystemConfiguration ? null, Foundation ? null
-}:
+{ lib, stdenv, fetchFromGitHub, SystemConfiguration ? null, Foundation ? null }:
```
Comment (SuperSandro2000) [inline pkgs/tools/typesetting/htmldoc/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, SystemConfiguration, Foundation }:
```

File: pkgs/tools/typesetting/htmldoc/default.nix [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
 
 assert stdenv.isDarwin -> SystemConfiguration != null
                        && Foundation != null;
 
 stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/typesetting/htmldoc/default.nix:6]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/tools/typesetting/htmldoc/default.nix [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
-      + "/release-${version}/htmldoc-${version}-source.tar.gz";
-    sha256 = "15x0xdf487j4i4gfap5yr83airxnbp2v4lxaz79a4s3iirrq39p0";
+  src = fetchFromGitHub {
+    owner = "michaelrsweet";
+    repo = "htmldoc";
+    rev = "v${version}";
+    sha256 = "0660829zjfdm6vzx14z7gvsfipsb7h0z74gbkyp2ncg3g2432s4n";
   };
   buildInputs = with stdenv;
        lib.optional isDarwin SystemConfiguration
```
Comment (SuperSandro2000) [inline pkgs/tools/typesetting/htmldoc/default.nix:16]:
```suggestion
  buildInputs = lib.optionals stdenv.isDarwin [ Foundation SystemConfiguration ]
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119226` at a2173042 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/9a729ec93f3b57c466dd097a227c59d9">htmldoc</a></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/typesetting/htmldoc/default.nix:22:5:
```
   |
22 |     license     = licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
gpl2Only or gpl2Plus? and platforms should be changed to unix.

File: pkgs/tools/typesetting/htmldoc/default.nix [hunk 4]
```diff
@@ -1,29 +1,23 @@
...
+    owner = "michaelrsweet";
+    repo = "htmldoc";
+    rev = "v${version}";
+    sha256 = "0660829zjfdm6vzx14z7gvsfipsb7h0z74gbkyp2ncg3g2432s4n";
   };
-  buildInputs = with stdenv;
-       lib.optional isDarwin SystemConfiguration
-    ++ lib.optional isDarwin Foundation;
+  buildInputs = [ zlib libpng ]
+      ++ lib.optionals stdenv.isDarwin [ Foundation SystemConfiguration ];
```
Comment (SuperSandro2000) [inline pkgs/tools/typesetting/htmldoc/default.nix:13]:
```suggestion
    ++ lib.optionals stdenv.isDarwin [ Foundation SystemConfiguration ];
```

File: pkgs/tools/typesetting/htmldoc/default.nix [hunk 5]
```diff
@@ -1,29 +1,23 @@
...
-}:
-
-assert stdenv.isDarwin -> SystemConfiguration != null
-                       && Foundation != null;
+{ lib, stdenv, fetchFromGitHub, zlib, libpng, SystemConfiguration, Foundation }:
 
 stdenv.mkDerivation rec {
-  version = "1.8.29";
+  version = "1.9.11";
   pname = "htmldoc";
```
Comment (SuperSandro2000) [inline pkgs/tools/typesetting/htmldoc/default.nix:5]:
```suggestion
  pname = "htmldoc";
  version = "1.9.11";
```

**Pull request merged**